### PR TITLE
Fixes #4516

### LINKED ISSUE #4516: MP Config TCK - Fix excluded tests
Currently, the below tests are excluded from execution when running the MP Config TCK. We need to either fix Quarkus so the test passes or challenge and fix the TCK if it's a problem there.

- [x] [ArrayConverterTest.*](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-config/tck-suite.xml#L17)
- [ ] [ConfigProviderTest.testInjectedConfigSerializable](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-config/tck-suite.xml#L26)
- [x] [ConfigProviderTest.testPropertyConfigSource](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-config/tck-suite.xml#L28)
- [x] [ConfigProviderTest.testEnvironmentConfigSource](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-config/tck-suite.xml#L36)

### PR DESCRIPTION
**Pull request by kenfinnigan**
- Ensures Config is serializable
- Passes Config serialization TCK test

### CODE REVIEW

**Comment by geoand** [comment]
I restarted the failing check

File: core/runtime/src/main/java/io/quarkus/runtime/configuration/AbstractDelegatingConfigSource.java [hunk 1]
```diff
@@ -12,9 +13,10 @@
 /**
  * A base class for configuration sources which delegate to other configuration sources.
  */
-public abstract class AbstractDelegatingConfigSource implements ConfigSource {
+public abstract class AbstractDelegatingConfigSource implements ConfigSource, Serializable {
+    private static final long serialVersionUID = -6636734120743034580L;
     protected final ConfigSource delegate;
-    private Map<String, String> propertiesMap;
+    protected Map<String, String> propertiesMap;
```
Comment (dmlloyd) [inline core/runtime/src/main/java/io/quarkus/runtime/configuration/AbstractDelegatingConfigSource.java]: What is the purpose of this change? Nothing new seems to use this field.

**Review by dmlloyd — APPROVED** [review]

File: core/runtime/src/main/java/io/quarkus/runtime/configuration/AbstractDelegatingConfigSource.java (hunk 1 above)
Comment (kenfinnigan) [reply core/runtime/src/main/java/io/quarkus/runtime/configuration/AbstractDelegatingConfigSource.java]: It might have been a carryover change from when I was trying to get serialization to work and needed access to it in the children.

Can revert back if needed

Comment (dmlloyd) [reply core/runtime/src/main/java/io/quarkus/runtime/configuration/AbstractDelegatingConfigSource.java]: If you wouldn't mind, that would be good.

Comment (kenfinnigan) [reply core/runtime/src/main/java/io/quarkus/runtime/configuration/AbstractDelegatingConfigSource.java]: done
